manifests/m/Microsoft/AzureCLI/2.53.1/Microsoft.AzureCLI.installer.yaml

```diff
@@ -4,9 +4,13 @@
 PackageIdentifier: Microsoft.AzureCLI
 PackageVersion: 2.53.1
 Installers:
-- Architecture: x64
+- Architecture: x86
   InstallerType: msi
   InstallerUrl: https://azcliprod.azureedge.net/msi/azure-cli-2.53.1.msi
   InstallerSha256: 57b8e2e4bf02059865eab97ba717ae576d9c45621be85b85045e4b1ed51c6efc
```

Comment:
```suggestion
  InstallerSha256: 57b8e2e4bf02059865eab97ba717ae576d9c45621be85b85045e4b1ed51c6efc
  ProductCode: '{8912DDBF-70DA-42F0-939A-278E4B42577D}'
```

Reply:
@mdanish-kh How do you get `ProductCode`?

Reply:
@bebound If you use one of the popular [manifest authoring tools](https://github.com/microsoft/winget-pkgs#authoring-a-manifest) to create the manifest, they will automatically add/update this field for you.

If you want to retrieve this for yourself, you can use a number of different methods:

- For installed MSI packages, you can get it by running the command (On PowerShell 5 / Windows PowerShell)

```
get-wmiobject Win32_Product | Sort-Object Name | Format-Table IdentifyingNumber, Name -AutoSize
```

- If you have the `.msi` file, you can use something like [MSI Viewer](https://apps.microsoft.com/detail/msi-viewer/9MZTR9QS01GN) or some PowerShell scripting to retrieve the ProductCode.

- You can run the [manifest in Windows Sandbox using the sandbox script](https://github.com/microsoft/winget-pkgs#in-windows-sandbox) and it will print out the related metadata for the package which includes the ProductCode

Reply:
I am able to get the `ProductCode` with the first method on my Windows machine (even though `Get-WmiObject Win32_Product` is slow):

```
> Get-WmiObject Win32_Product | Sort-Object Name | Format-Table IdentifyingNumber, Name, Version -AutoSize

IdentifyingNumber                      Name                                                           Version
-----------------                      ----                                                           -------
...
{C56C2883-5C38-4789-B6BA-4D4B17440AE6} Microsoft Azure CLI (64-bit)                                   2.54.0
```

All these methods require either installing the MSI or other dependencies. As @azclibot is currently hosted on a Linux agent, these methods may not be practicable unless we move @azclibot to a Windows agent.

I am not sure if `ProductCode` comes from `Product.Id`:

https://github.com/Azure/azure-cli/blob/aeb94ed12a46acdd14e7c4e2633908e149f4a1bc/build_scripts/windows/Product.wxs#L36

```
  <Product Id="*"
```

If so, is it possible to have a fixed GUID, instead of generating a new one for each MSI package? cc @heaths

Reply:
As this conversation has been closed, I commented directly on the PR: https://github.com/microsoft/winget-pkgs/pull/123641#issuecomment-1814363236